test/e2e/automigration/automigration.go

```diff
@@ -228,4 +239,353 @@ var _ = ginkgo.Describe("auto migration", autoMigrationTestLabel, func() {
+				ctx, overridePolicyForDp2, metav1.CreateOptions{},
+			)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
+		})
+
+		// 5. Set policies
+
+		policies.SetPropagationPolicy(dp1, propPolicy)
+		policies.SetOverridePolicy(dp1, overridePolicyForDp1.Name)
+		policies.SetPropagationPolicy(dp2, propPolicy)
+		policies.SetOverridePolicy(dp2, overridePolicyForDp2.Name)
+
+		// 6. Create deployments and check results
+
+		ginkgo.By("Creating original deployment", func() {
+			dp1, err = f.HostKubeClient().AppsV1().Deployments(f.TestNamespace().Name).Create(
+				ctx, dp1, metav1.CreateOptions{},
+			)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
+		})
+
+		// This function checks a deployment for two things:
+		// 1) the replica count for clusterToMigrateFrom remains the same as replicasPerCluster
+		// 2) the replica counts for other clusters match an expected distribution given by newReplicaCounts.
+		checkPodsMigrated := func(dpName string, clusterToMigrateFrom *fedcorev1a1.FederatedCluster, newReplicaCounts map[string]int32) {
+			ginkgo.GinkgoLogr.Info(fmt.Sprintf("expected distribution: %v", newReplicaCounts))
+			ctx, cancel := context.WithTimeout(ctx, resourcePropagationTimeout+autoMigrationTimeout)
+			defer cancel()
+
+			for _, cluster := range clusters {
+				if cluster.Name == clusterToMigrateFrom.Name {
+					continue
+				}
+
+				gomega.Eventually(ctx, func(g gomega.Gomega) {
+					clusterDp, err := f.ClusterKubeClient(ctx, cluster).AppsV1().Deployments(f.TestNamespace().Name).Get(
+						ctx, dpName, metav1.GetOptions{},
+					)
+					gomega.Expect(err).To(gomega.Or(gomega.Not(gomega.HaveOccurred()), gomega.Satisfy(apierrors.IsNotFound)))
+					g.Expect(err).ToNot(gomega.HaveOccurred())
+					g.Expect(clusterDp.Spec.Replicas).To(gomega.HaveValue(gomega.Equal(newReplicaCounts[cluster.Name])))
+				}).WithPolling(defaultPollingInterval).Should(gomega.Succeed())
+			}
+
+			// This ensures keepUnschedulabeReplicas is working correctly
+			gomega.Consistently(ctx, func(g gomega.Gomega) {
+				clusterDp, err := f.ClusterKubeClient(ctx, clusterToMigrateFrom).AppsV1().Deployments(f.TestNamespace().Name).Get(
+					ctx, dp2.Name, metav1.GetOptions{},
+				)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+				g.Expect(clusterDp.Spec.Replicas).To(gomega.HaveValue(gomega.Equal(replicasPerCluster)))
+			}).WithTimeout(5 * time.Second).Should(gomega.Succeed())
+		}
+
+		expectedDistributionDp1 := map[string]int32{}
+		for _, cluster := range clusters {
+			if cluster.Name == clusterToMigrateFromForDp1.Name {
+				continue
+			}
+
+			// The deployment in clusterToMigrateFromForDp1 will only have 1 schedulable pod. The remaining should be
+			// migrated to the remaining clusters evenly (based on the propagation policy).
+			expectedDistributionDp1[cluster.Name] = replicasPerCluster + ((replicasPerCluster)-1)/int32(len(clusters)-1)
```

Comment:
This division only works when there is no remainder (such as this case). Can we move this closer to where `replicasPerCluster` is defined so we can change them in tandem if required.